src/middleware/auth.ts

```diff
@@ -26,7 +26,7 @@ const authMiddleWare = async (
       res.status(401).send({ message: 'Invalid token: User not found' });
       return;
     }
-    req.user = user;
+    req.user = { id: user.id };
```

Comment:
You should also add user `name `and `email `here as we need it in user profile, just don't add `password`.

Reply:
@ahzia, it seems also to me, that adding only the user ID to the request is proper because the request data can be accessed in every other controller that would need only the user ID and not the other user data. Then, in the profile, we can retrieve the user data through the orm method and filter out the password. Also, at some point later, the user will have more data to display in the profile, update it, etc, what does not seem right to add it all to the request.

Reply:
@Francieli-S 
@bojana-petroska 
We are retrieving users details ones in the middleware, there is no need to retrieve them again; we should avoid sending unnecessary requests to database as it adds up lots of loads.